lib/libutils/isoc/arch/arm/arm32_aeabi_softfloat.c

```diff
@@ -25,47 +25,57 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include <compiler.h>
 #include "platform.h"
 #include <softfloat.h>
 
 /*
- * Helpers to convert between float32 and float, and float64 and double
- * used by the AEABI functions below.
+ * The AEABI functions uses soft-float calling convention even if the
+ * functions are compiled for hard-float. So where float and double would
+ * have been expected we use aeabi_float_t and aeabi_double_t respectively
+ * instead.
```

Comment:
Could you replace this comment (that I find unclear) with the very-clear description how soft and hard float are used in arm32 you gave in the offline email?